README.md

```diff
@@ -17,83 +17,318 @@ Main characteristics:
-  return quantity() > 0 ? quantity() * unitPrice() : 0
+  return () => {
+    console.log('No more subscriber');
+  }
+});
+
+const unsubscribe = values$.subscribe((value) => {
+  console.log(`value = ${value}`);
 });
-totalPrice.subscribe((totalPrice) => console.log(totalPrice)); // logs any change to totalPrice
-quantity.set(0);
-unitPrice.set(20);
+
+value$.set(1);
+unsubscribe();
+```
+
+output:
+
+```text
+  At least one subscriber
+  value = 0
+  value = 1
+  No more subscriber
 ```
 
-Note that every store created with `store = writable(value)` is a function that returns the value of the store. It is equivalent to `get(store)`.
-When getting the value of a store (either by calling the store as a function or by using `get`) from a reactive context (such as the one created by `computed`) the dependency on that store is recorded and any change to that store will trigger a recomputation.
+### derived
+
+[api documentation](https://amadeusitgroup.github.io/tansu/functions/derived.html)
+
+A `derived` store calculates its value based on one or more other stores provided as parameters.
+Since its value is derived from other stores, it is a read-only store and does not have any `set` or `update` methods.
 
-The output of this example will be:
+#### Single store
+
+```typescript
+import {writable, derived} from "@amadeus-it-group/tansu";
+
+const value$ = writable(1);
+const double$ = derived(value$, (value) => value * 2);
+
+double$.subscribe((double) => console.log('Double value', double));
+value$.set(2);
+```
+
+output:
 
 ```text
-computing the total price
-20
-computing the total price
-0
+Double value 2
+Double value 4
 ```
 
-Note that when the quantity is 0, changes to the unit price no longer trigger an update of the total price because the list of dependencies is dynamic (and `unitPrice()` is not called if `quantity()` is `0`).
+#### Multiple stores
+
+```typescript
+import {writable, derived} from "@amadeus-it-group/tansu";
 
-### A batch function is available
+const a$ = writable(1);
+const b$ = writable(1);
+const sum$ = derived([a$, b$], ([a, b]) => a + b);
+
+sum$.subscribe((sum) => console.log('Sum', sum));
+a$.set(2);
+```
 
-Depending on multiple stores can lead to some issues. Let's have a look at the following example:
+output:
+
+```text
+Sum 2
+Sum 3
+```
+
+#### Asynchronous set
+
+A `derived` can directly manipulate its value using the set method instead of relying on the returned value of the provided function.
+This flexibility allows you to manage asynchronous operations or apply filtering logic before updating the observable's value.
 
 ```typescript
-import {writable, derived} from 'svelte/store';
+import {writable, derived} from "@amadeus-it-group/tansu";
+
+const a$ = writable(0);
+const asynchronousDouble$ = derived(a$, (a, set) => {
+  setTimeout(() => set(a * 2));
+});
+
+const evenOnly$ = derived(a$, (a, set) => {
+  if (a !== undefined && a % 2 === 0) {
+      set(a);
+  }
+}, <number>undefined);
```

Comment:
I don't think it is a good idea to show an example that lies about the type of `evenOnly$`. If it can be initially `undefined`, this should be visible in the type.

```suggestion
}, undefined as number | undefined);
```